flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/DispatcherCachedOperationsHandler.java

```diff
@@ -39,30 +40,66 @@ public class DispatcherCachedOperationsHandler {
     private final CompletedOperationCache<AsynchronousJobOperationKey, String>
             savepointTriggerCache;
 
+    private final CompletedOperationCache<AsynchronousJobOperationKey, Long> checkpointTriggerCache;
+
+    private final TriggerCheckpointFunction triggerCheckpointFunction;
+
     private final TriggerSavepointFunction triggerSavepointFunction;
 
     private final TriggerSavepointFunction stopWithSavepointFunction;
 
     DispatcherCachedOperationsHandler(
             DispatcherOperationCaches operationCaches,
+            TriggerCheckpointFunction triggerCheckpointFunction,
             TriggerSavepointFunction triggerSavepointFunction,
             TriggerSavepointFunction stopWithSavepointFunction) {
         this(
+                triggerCheckpointFunction,
+                operationCaches.getCheckpointTriggerCache(),
                 triggerSavepointFunction,
                 stopWithSavepointFunction,
                 operationCaches.getSavepointTriggerCache());
     }
 
     @VisibleForTesting
     DispatcherCachedOperationsHandler(
+            TriggerCheckpointFunction triggerCheckpointFunction,
+            CompletedOperationCache<AsynchronousJobOperationKey, Long> checkpointTriggerCache,
             TriggerSavepointFunction triggerSavepointFunction,
             TriggerSavepointFunction stopWithSavepointFunction,
             CompletedOperationCache<AsynchronousJobOperationKey, String> savepointTriggerCache) {
+        this.triggerCheckpointFunction = triggerCheckpointFunction;
+        this.checkpointTriggerCache = checkpointTriggerCache;
         this.triggerSavepointFunction = triggerSavepointFunction;
         this.stopWithSavepointFunction = stopWithSavepointFunction;
         this.savepointTriggerCache = savepointTriggerCache;
     }
 
+    public CompletableFuture<Acknowledge> triggerCheckpoint(
+            AsynchronousJobOperationKey operationKey,
+            CheckpointBackupType checkpointBackupType,
+            Time timeout) {
+
+        if (!checkpointTriggerCache.containsOperation(operationKey)) {
+            checkpointTriggerCache.registerOngoingOperation(
+                    operationKey,
+                    triggerCheckpointFunction.apply(
+                            operationKey.getJobId(), checkpointBackupType, timeout));
+        }
+
+        return CompletableFuture.completedFuture(Acknowledge.get());
+    }
+
+    public CompletableFuture<OperationResult<Long>> getCheckpointStatus(
```

Comment:
The result type is `Long`. what does the long value result indicate? I assume the `Long` type is not meant for checkpoint Id.

Reply:
I got it now after seeing the other discussion. it is for checkpoint id. For savepoint trigger, the result is a path/location. for checkpoint trigger, the result is a checkpoint id.